packages/ld-key-pair/KeyPair.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * defines the ability to sign based on a KeyPair or only
+ * from a private key
+ */
+interface Signer {
+    privateKey: string;
+    keyPair?: KeyPair;
+    sign(options: {
+        data: any;
+    }): string;
+}
+
+/**
+ * defines the ability to perform verification based on a KeyPair
+ * or a public key
+ */
+interface Verifier {
+    publicKey: string;
+    keyPair?: KeyPair;
+    verify(options: {
+        data: any;
+        signature: string;
+    }): boolean;
+}
+
+/**
+ * public key side of a key pair
+ */
+interface PublicNode {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    publicKeyBase58?: string;
+}
+/**
+ * public key side of a key pair
+ */
+ interface PrivateNode {
+    id: string;
+    type: any;
+    controller?: any;
+    privateKey?: string;
+    privateKeyBase58?: string;
+}
+
+/**
+ * defines the common properties and methoods in use by 
+ * key pairs in the LD realm based on existing implementations
+ */
+interface KeyPair {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    privateKey?: string;
+    publicKeyBase58?: string;
+    privateKeyBase58?: string;
+    publicNode(): PublicNode;
```

Comment:
I am not sure we need `publicNode`... prefer to remove it until we see it needed

Reply:
added this to ensure compatability with [DB](https://github.com/digitalbazaar/ecdsa-secp256k1-verification-key-2019/blob/master/lib/index.js) and others - public/private nodes can be handy to prevent info leakage, but can go either way on this - basically just a means of exposing half a pair and related functionality and properties

Reply:
e.g. let's us drop in full compatability with [db secp256k1](https://github.com/digitalbazaar/ecdsa-secp256k1-verification-key-2019/blob/c0385a53d23f4fc3f038ea5ce159f84c3f123f6b/lib/index.js#L218) right out of the gate

Reply:
its semi redundant to "toJsonWebKey(false) or toKeyPair(false)" both return a public node in jwk or base58... this method assumes you always return base58.... bad assumption.